pkg/pillar/hypervisor/kvm.go

```diff
@@ -707,6 +726,44 @@ func vmmOverhead(domainName string, domainUUID uuid.UUID, domainRAMSize int64, v
 	return overhead, nil
 }
 
+func getOVMFSettingsFile(domainName string) string {
+	return types.OVMFSettingsDir + "/" + domainName + "_OVMF_VARS.fd"
+}
+
+func prepareOVMFSettings(domainName string) (string, error) {
+	logrus.Infof("@ohm: I Preparing OVMF settings for domain %s", domainName)
+
+	// Create the OVMF settings directory if it does not exist
+	if _, err := os.Stat(types.OVMFSettingsDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(types.OVMFSettingsDir, 0755); err != nil {
+			return "", logError("failed to create OVMF settings directory: %v", err)
+		}
+	}
```

Comment:
```suggestion
		if err := os.MkdirAll(types.OVMFSettingsDir, 0755); err != nil {
			return "", logError("failed to create OVMF settings directory: %v", err)
		}
```

> If path is already a directory, MkdirAll does nothing and returns nil.

or is it the case if it is a file, then it is ok?

Reply:
No, it shouldn't, like `mkdir -p`.

Reply:
Ok, will change it, thanks!

Reply:
Fixed